core/src/main/java/com/klaytn/caver/abi/ABI.java

```diff
@@ -0,0 +1,317 @@
+        String functionSignature = ABI.buildFunctionSignature(method);
+
+        List<String> solTypeList = new ArrayList();
+
+        for (ContractIOType contractIOType : method.getInputs()) {
+            solTypeList.add(contractIOType.getType());
+        }
+
+        return encodeFunctionCall(functionSignature, solTypeList, params);
+    }
+
+    /**
+     * Encodes a function call.
+     * @param functionSig A function signature string.
+     * @param params A List of method parameter.
+     * @return String
+     */
+    public static String encodeFunctionCall(String functionSig, List<String> solTypeList, List<Object> params) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        String methodId = encodeFunctionSignature(functionSig);
+        String encodedParams = encodeParameters(solTypeList, params);
+
+        return methodId + encodedParams;
+    }
+
+    /**
+     * Encodes a function signature.
+     * @param method A ContractMethod instance.
+     * @return String
+     */
+    public static String encodeFunctionSignature(ContractMethod method) {
+        return encodeFunctionSignature(buildFunctionSignature(method));
+    }
+
+    /**
+     * Encodes a function signature.
+     * @param functionSig A function signature string.
+     * @return String
+     */
+    public static String encodeFunctionSignature(String functionSig) {
+        byte[] input = functionSig.getBytes();
+        byte[] hash = Hash.sha3(input);
+        return Numeric.toHexString(hash).substring(0, 10);
+    }
+
+    /**
+     * Build a function signature.
+     * @param method A ContractMethod instance.
+     * @return String
+     */
+    public static String buildFunctionSignature(ContractMethod method) {
+        StringBuilder result = new StringBuilder();
+        result.append(method.getName());
+        result.append("(");
+        String params = method.getInputs().stream()
+                .map(ContractIOType::getType)
+                .collect(Collectors.joining(","));
+        result.append(params);
+        result.append(")");
+
+        return result.toString();
+    }
+
+    /**
+     * Encodes a event signature.
+     * @param event A ContractEvent instance.
+     * @return String
+     */
+    public static String encodeEventSignature(ContractEvent event) {
+        return encodeEventSignature(buildEventSignature(event));
+    }
+
+    /**
+     * Encodes a event signature.
+     * @param eventSig A event signature.
+     * @return String
+     */
+    public static String encodeEventSignature(String eventSig) {
+        byte[] input = eventSig.getBytes();
+        byte[] hash = Hash.sha3(input);
+        return Numeric.toHexString(hash);
+    }
+
+    /**
+     * Build a event signature.
+     * @param event A ContractEvent instance
+     * @return String
+     */
+    public static String buildEventSignature(ContractEvent event) {
```

Comment:
How about changing this function name to `buildEventString` or something else?

Reply:
Okay.